### DISCUSSION TITLE
Webpack alteres web worker file

### DISCUSSION THREAD

**Discussion by directcuteo** [webpack/webpack#17885] [Q&A]
I am trying to use an external npm library in my Angular project that relies on web workers and `wasm`. The issue is that webpack does something, it changes the code and it does so that it becomes invalid (throwing an error).

![image](https://github.com/webpack/webpack/assets/37619567/1803cb95-3df3-4bb0-867e-583c36eebe66)
![image](https://github.com/webpack/webpack/assets/37619567/d03610d1-e4c2-4786-8110-cc67277218b4)

And here is that function's body in its original form(in the IDE)
![image](https://github.com/webpack/webpack/assets/37619567/37bdd33c-d764-47ee-b9ae-dbdd61a0eb56)

I don't know why there is a `_mainWorker` function but that one is not found anywhere in the code.

The file is grabbed at runtime as blob using http request.

[a comment by an automated account is not shown]

**Comment by alexander-akait** [comment]
Please create reproducible test repo, thank you
